### ISSUE TITLE
platformio_options (board freq) not usable on firebeetle32 after update to 2024.6.0

### ISSUE THREAD

**Issue by emefff** [esphome/issues#5929]
### The problem

This issue arose during investigating issue https://github.com/esphome/issues/issues/5924 . 
Essentially, when a firebeetl32 is used, the board frequency cannot be chosen anymore in 2024.6.0.
This, in turn, means, that users of battery powered devices, for which firebeetle32 is basically the NR.1 option, will experience shortened runtimes of their devices with 2024.6+. If lower freqs cannot be used with firebeetle32, there's almost no point using it in battery powered devices.
The freq-option still works on esp32dev devices, only physical firebeetle32 devices are affected. For the physical firebeetle32, it does not seem to matter if in the .yaml board:=firebeetle32 or esp32dev.

### Which version of ESPHome has the issue?

2024.6.0

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2024.06

### What platform are you using?

ESP32

### Board

firebeetle32

### Component causing the issue

option platformio_options board_build.f_cpu: must not be used

### Example YAML snippet

```yaml
esphome:
  name: esphome--12
  platformio_options:
    board_build.f_cpu: 81000000L #board temp lower, cannot use 80000000L because of issue 3909

esp32:
  board: firebeetle32
  framework:
    type: arduino
```

### Anything in the logs that might be useful for us?

```txt
*** device enters boot-loop when platformio_options: board_build.f_cpu: ANYNUMBER L
*** is given. Only commenting leads to a functioning FW-file.

*** excerpt from the log, unsuccessful boot attempts before are not shown (length!)
**********************************************************************************
[13:56:59]Guru Meditation Error: Core  1 panic'ed (Interrupt wdt timeout on CPU1). 
[13:56:59]
[13:56:59]Core  1 register dump:
[13:56:59]PC      : 0x4008fde4  PS      : 0x00060435  A0      : 0x800813c4  A1      : 0x3ffb2390  
[13:56:59]A2      : 0x00000001  A3      : 0x3ffbf0b8  A4      : 0x00000019  A5      : 0x00000000  
[13:56:59]A6      : 0x00060e23  A7      : 0x00000001  A8      : 0x8008f0cb  A9      : 0x3ffb2370  
[13:56:59]A10     : 0x3ffbf0b8  A11     : 0x0000017a  A12     : 0x0000017a  A13     : 0x00060423  
[13:56:59]A14     : 0x00060423  A15     : 0x00000001  SAR     : 0x00000019  EXCCAUSE: 0x00000006  
[13:56:59]EXCVADDR: 0x00000000  LBEG    : 0x40084951  LEND    : 0x40084959  LCOUNT  : 0x00000027  
[13:56:59]
[13:56:59]
[13:56:59]Backtrace:0x4008fde1:0x3ffb23900x400813c1:0x3ffb23b0 0x400e1072:0x3ffb23d0 0x400813dd:0x3ffb23f0 0x40081577:0x3ffb2410 0x4008161d:0x3ffb2430 0x400daeb2:0x3ffb2460 0x400d9abd:0x3ffb24b0 0x400d9aca:0x3ffb24d0 0x40171e79:0x3ffb24f0 0x40171f05:0x3ffb2510 0x400e0414:0x3ffb2530 0x400e3825:0x3ffb2560 0x400ee782:0x3ffb2820 
[13:56:59]
[13:56:59]
[13:56:59]Core  0 register dump:
[13:56:59]PC      : 0x40173ada  PS      : 0x00060935  A0      : 0x800f76a8  A1      : 0x3ffbd500  
[13:56:59]A2      : 0x00000000  A3      : 0x40089020  A4      : 0x00060920  A5      : 0x3ffbca80  
[13:56:59]A6      : 0x007bf0b8  A7      : 0x003fffff  A8      : 0x800f717a  A9      : 0x3ffbd4d0  
[13:56:59]A10     : 0x00000000  A11     : 0x3ffbf0b4  A12     : 0x3ffbf0b4  A13     : 0x00000000  
[13:56:59]A14     : 0x00060920  A15     : 0x00000000  SAR     : 0x0000001a  EXCCAUSE: 0x00000006  
[13:56:59]EXCVADDR: 0x00000000  LBEG    : 0x00000000  LEND    : 0x00000000  LCOUNT  : 0x00000000  
[13:56:59]
[13:56:59]
[13:56:59]Backtrace:0x40173ad7:0x3ffbd5000x400f76a5:0x3ffbd520 0x4008dfd4:0x3ffbd540 
[13:56:59]
[13:56:59]
[13:56:59]
[13:56:59]
[13:56:59]ELF file SHA256: 0000000000000000
[13:56:59]
[13:56:59]Rebooting...
[13:56:59]ets Jun  8 2016 00:22:57
[13:56:59]
[13:56:59]rst:0xc (SW_CPU_RESET),boot:0x1b (SPI_FAST_FLASH_BOOT)
[13:56:59]configsip: 0, SPIWP:0xee
[13:56:59]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[13:56:59]mode:DIO, clock div:2
[13:56:59]load:0x3fff0018,len:4
[13:56:59]load:0x3fff001c,len:1044
[13:56:59]load:0x40078000,len:8896
[13:56:59]load:0x40080400,len:5828
[13:56:59]entry 0x400806ac
[13:57:00][I][logger:156]: Log initialized
[13:57:00][C][status_led:014]: Setting up Status LED...
[13:57:00][C][safe_mode:079]: There have been 7 suspected unsuccessful boot attempts
[13:57:00][D][esp32.preferences:114]: Saving 1 preferences to flash...
[13:57:00][D][esp32.preferences:143]: Saving 1 preferences to flash: 0 cached, 1 written, 0 failed
[13:57:00][I][app:029]: Running through setup()...
[13:57:00][C][gpio.one_wire:011]: Setting up 1-wire bus...
[13:57:00][C][gpio.one_wire:011]: Setting up 1-wire bus...
[13:57:00][C][gpio.one_wire:011]: Setting up 1-wire bus...
[13:57:01]ets Jun  8 2016 00:22:57
[13:57:01]
[13:57:01]rst:0x8 (TG1WDT_SYS_RESET),boot:0x1b (SPI_FAST_FLASH_BOOT)
[13:57:01]configsip: 0, SPIWP:0xee
[13:57:01]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[13:57:01]mode:DIO, clock div:2
[13:57:01]load:0x3fff0018,len:4
[13:57:01]load:0x3fff001c,len:1044
[13:57:01]load:0x40078000,len:8896
[13:57:01]load:0x40080400,len:5828
[13:57:01]entry 0x400806ac
[13:57:01][I][logger:156]: Log initialized
[13:57:01][C][status_led:014]: Setting up Status LED...
[13:57:01][C][safe_mode:079]: There have been 8 suspected unsuccessful boot attempts
[13:57:01][D][esp32.preferences:114]: Saving 1 preferences to flash...
[13:57:01][D][esp32.preferences:143]: Saving 1 preferences to flash: 0 cached, 1 written, 0 failed
[13:57:01][I][app:029]: Running through setup()...
[13:57:01][C][gpio.one_wire:011]: Setting up 1-wire bus...
[13:57:01][C][gpio.one_wire:011]: Setting up 1-wire bus...
[13:57:01][C][gpio.one_wire:011]: Setting up 1-wire bus...
[13:57:02]Guru Meditation Error: Core  1 panic'ed (Interrupt wdt timeout on CPU1). 
[13:57:02]
[13:57:02]Core  1 register dump:
[13:57:02]PC      : 0x4008e0cc  PS      : 0x00060435  A0      : 0x8008f0c6  A1      : 0x3ffb2370  
[13:57:02]A2      : 0x3ffc47bc  A3      : 0x0000017a  A4      : 0x3ffc4598  A5      : 0x00060423  
[13:57:02]A6      : 0x00060423  A7      : 0x00000001  A8      : 0x3ffb88d4  A9      : 0x3ffb2350  
[13:57:02]A10     : 0x3ffc45b0  A11     : 0x00060423  A12     : 0x00060423  A13     : 0x3f401ee0  
[13:57:02]A14     : 0x007bf0b8  A15     : 0x003fffff  SAR     : 0x00000019  EXCCAUSE: 0x00000006  
[13:57:02]EXCVADDR: 0x00000000  LBEG    : 0x40084951  LEND    : 0x40084959  LCOUNT  : 0x00000027  
[13:57:02]
[13:57:02]
[13:57:02]Backtrace:0x4008e0c9:0x3ffb23700x4008f0c3:0x3ffb2390 0x400813c1:0x3ffb23b0 0x400e1072:0x3ffb23d0 0x400813dd:0x3ffb23f0 0x40081577:0x3ffb2410 0x4008161d:0x3ffb2430 0x400daeb2:0x3ffb2460 0x400d9abd:0x3ffb24b0 0x400d9aca:0x3ffb24d0 0x40171e79:0x3ffb24f0 0x40171f05:0x3ffb2510 0x400e0414:0x3ffb2530 0x400e3825:0x3ffb2560 0x400ee782:0x3ffb2820 
[13:57:02]
[13:57:02]
[13:57:02]Core  0 register dump:
[13:57:02]PC      : 0x40173ada  PS      : 0x00060935  A0      : 0x800f76a8  A1      : 0x3ffbd500  
[13:57:02]A2      : 0x00000000  A3      : 0x40089020  A4      : 0x00060920  A5      : 0x3ffbc990  
[13:57:02]A6      : 0x007bf0b8  A7      : 0x003fffff  A8      : 0x800f717a  A9      : 0x3ffbd4d0  
[13:57:02]A10     : 0x00000000  A11     : 0x3ffbf0b4  A12     : 0x3ffbf0b4  A13     : 0x00000000  
[13:57:02]A14     : 0x00060920  A15     : 0x00000000  SAR     : 0x0000001a  EXCCAUSE: 0x00000006  
[13:57:02]EXCVADDR: 0x00000000  LBEG    : 0x00000000  LEND    : 0x00000000  LCOUNT  : 0x00000000  
[13:57:02]
[13:57:02]
[13:57:02]Backtrace:0x40173ad7:0x3ffbd5000x400f76a5:0x3ffbd520 0x4008dfd4:0x3ffbd540 
[13:57:02]
[13:57:02]
[13:57:02]
[13:57:02]
[13:57:02]ELF file SHA256: 0000000000000000
[13:57:02]
[13:57:02]Rebooting...
[13:57:02]ets Jun  8 2016 00:22:57
[13:57:02]
[13:57:02]rst:0xc (SW_CPU_RESET),boot:0x1b (SPI_FAST_FLASH_BOOT)
[13:57:02]configsip: 0, SPIWP:0xee
[13:57:02]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[13:57:02]mode:DIO, clock div:2
[13:57:02]load:0x3fff0018,len:4
[13:57:02]load:0x3fff001c,len:1044
[13:57:02]load:0x40078000,len:8896
[13:57:02]load:0x40080400,len:5828
[13:57:02]entry 0x400806ac
[13:57:02][I][logger:156]: Log initialized
[13:57:02][C][status_led:014]: Setting up Status LED...
[13:57:02][C][safe_mode:079]: There have been 9 suspected unsuccessful boot attempts
[13:57:02][D][esp32.preferences:114]: Saving 1 preferences to flash...
[13:57:02][D][esp32.preferences:143]: Saving 1 preferences to flash: 0 cached, 1 written, 0 failed
[13:57:02][I][app:029]: Running through setup()...
[13:57:02][C][gpio.one_wire:011]: Setting up 1-wire bus...
[13:57:02][C][gpio.one_wire:011]: Setting up 1-wire bus...
[13:57:02][C][gpio.one_wire:011]: Setting up 1-wire bus...
[13:57:03]ets Jun  8 2016 00:22:57
[13:57:03]
[13:57:03]rst:0x8 (TG1WDT_SYS_RESET),boot:0x1b (SPI_FAST_FLASH_BOOT)
[13:57:03]configsip: 0, SPIWP:0xee
[13:57:03]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[13:57:03]mode:DIO, clock div:2
[13:57:03]load:0x3fff0018,len:4
[13:57:03]load:0x3fff001c,len:1044
[13:57:03]load:0x40078000,len:8896
[13:57:03]load:0x40080400,len:5828
[13:57:03]entry 0x400806ac
[13:57:03][I][logger:156]: Log initialized
[13:57:03][C][status_led:014]: Setting up Status LED...
[13:57:03][C][safe_mode:079]: There have been 10 suspected unsuccessful boot attempts
[13:57:03][D][esp32.preferences:114]: Saving 1 preferences to flash...
[13:57:03][D][esp32.preferences:143]: Saving 1 preferences to flash: 0 cached, 1 written, 0 failed
[13:57:03][E][safe_mode:086]: Boot loop detected. Proceeding to safe mode
[13:57:03][E][component:164]: Component safe_mode set Error flag: unspecified
[13:57:04][I][app:029]: Running through setup()...
[13:57:04][C][wifi:047]: Setting up WiFi...
[13:57:04][C][wifi:060]: Starting WiFi...
[13:57:04][C][wifi:061]:   Local MAC: ***REDACTED***
[13:57:04][I][wifi:312]: WiFi Connecting to 'giedi_prime_nomap'...
[13:57:04][W][component:157]: Component wifi set Warning flag: associating to network
[13:57:09][I][wifi:616]: WiFi Connected!
[13:57:09][C][wifi:427]:   Local MAC: ***REDACTED***
[13:57:09][C][wifi:432]:   SSID: 'giedi_prime_nomap'
[13:57:09][C][wifi:435]:   IP Address: ***REDACTED***
[13:57:09][C][wifi:439]:   BSSID: ***REDACTED***
[13:57:09][C][wifi:440]:   Hostname: 'esphome--12'
[13:57:09][C][wifi:442]:   Signal strength: -37 dB ▂▄▆█
[13:57:09][C][wifi:446]:   Channel: 1
[13:57:09][C][wifi:447]:   Subnet: 255.255.255.0
[13:57:09][C][wifi:448]:   Gateway: ***REDACTED***
[13:57:09][C][wifi:449]:   DNS1: 0.0.0.0
[13:57:09][C][wifi:450]:   DNS2: 0.0.0.0
[13:57:09][D][wifi:625]: Disabling AP...
[13:57:09][D][wifi:775]: Saved fast_connect wifi settings
[13:57:09][I][app:062]: setup() finished successfully!
[13:57:09][W][safe_mode:099]: SAFE MODE IS ACTIVE
[13:57:09][W][component:170]: Component wifi cleared Warning flag
[13:57:09][I][app:100]: ESPHome version 2024.6.0 compiled on Jun 20 2024, 13:55:55
[13:57:09][C][status_led:019]: Status LED:
[13:57:09][C][status_led:020]:   Pin: GPIO2
[13:57:09][C][wifi:599]: WiFi:
[13:57:09][C][wifi:427]:   Local MAC: ***REDACTED***
[13:57:09][C][wifi:432]:   SSID: 'giedi_prime_nomap'
[13:57:09][C][wifi:435]:   IP Address: ***REDACTED***
[13:57:09][C][wifi:439]:   BSSID: ***REDACTED***
[13:57:09][C][wifi:440]:   Hostname: 'esphome--12'
[13:57:09][C][wifi:442]:   Signal strength: -38 dB ▂▄▆█
[13:57:09][C][wifi:446]:   Channel: 1
[13:57:09][C][wifi:447]:   Subnet: 255.255.255.0
[13:57:09][C][wifi:448]:   Gateway: ***REDACTED***
[13:57:09][C][wifi:449]:   DNS1: 0.0.0.0
[13:57:09][C][wifi:450]:   DNS2: 0.0.0.0
[13:57:09][C][logger:185]: Logger:
[13:57:09][C][logger:186]:   Level: DEBUG
[13:57:09][C][logger:188]:   Log Baud Rate: 115200
[13:57:09][C][logger:189]:   Hardware UART: UART0
[13:57:09][C][captive_portal:088]: Captive Portal:
[13:57:09][C][mdns:115]: mDNS:
[13:57:09][C][mdns:116]:   Hostname: esphome--12
[13:57:09][C][esphome.ota:073]: Over-The-Air updates:
[13:57:09][C][esphome.ota:074]:   Address: ***REDACTED***:3232
[13:57:09][C][esphome.ota:075]:   Version: 2
[13:57:09][C][esphome.ota:078]:   Password configured
[13:57:09][C][safe_mode:018]: Safe Mode:
[13:57:09][C][safe_mode:020]:   Boot considered successful after 60 seconds
[13:57:09][C][safe_mode:021]:   Invoke after 10 boot attempts
[13:57:09][C][safe_mode:023]:   Remain in safe mode for 300 seconds
[13:57:09][W][safe_mode:031]: SAFE MODE IS ACTIVE
[13:58:03][I][safe_mode:041]: Boot seems successful; resetting boot loop counter
[13:58:03][D][esp32.preferences:114]: Saving 2 preferences to flash...
[13:58:03][D][esp32.preferences:143]: Saving 2 preferences to flash: 1 cached, 1 written, 0 failed
```

### Additional information

As soon as I set:

esphome:
  name: esphome--12
  #platformio_options:
    #board_build.f_cpu: 81000000L #board temp lower, cannot use 80000000L because of issue 3909

esp32:
  board: firebeetle32 #esp32dev also works
  framework:
    type: arduino

it works again.

**Comment by emefff** [comment]
Switching back to dallasng solves this problem.

**Comment by emefff** [comment]
This issue seems resolved with updating to 2024.6.3. One_wire dallas_temp is working as advertised now.
